src/main.cpp

```diff
@@ -6019,10 +6024,9 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
     else if (strCommand == NetMsgType::BLOCK && !fImporting && !fReindex) // Ignore blocks received while importing
     {
         CBlock block;
+        pfrom->nBlockSize = vRecv.size(); // Used by ProcessNewBlock() debug
```

Comment:
That's not needed, and it's a layer violation, as this is the *serialized* size, not the block size.